pkg/asset/manifests/infrastructure_test.go

```diff
@@ -0,0 +1,181 @@
+		),
+		expectedInfrastructure: infraBuild.build(
+			infraBuild.forPlatform(configv1.AWSPlatformType),
+			infraBuild.withServiceEndpoint("service", "https://endpoint"),
+		),
+	}}
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			parents := asset.Parents{}
+			parents.Add(
+				&installconfig.ClusterID{
+					UUID:    "test-uuid",
+					InfraID: "test-infra-id",
+				},
+				&installconfig.InstallConfig{Config: tc.installConfig},
+				&CloudProviderConfig{},
+				&AdditionalTrustBundleConfig{},
+			)
+			infraAsset := &Infrastructure{}
+			err := infraAsset.Generate(parents)
+			if !assert.NoError(t, err, "failed to generate asset") {
+				return
+			}
+			if !assert.Len(t, infraAsset.FileList, 1, "expected only one file to be generated") {
+				return
+			}
+			assert.Equal(t, infraAsset.FileList[0].Filename, "manifests/cluster-infrastructure-02-config.yml")
+			var actualInfra configv1.Infrastructure
+			err = yaml.Unmarshal(infraAsset.FileList[0].Data, &actualInfra)
+			if !assert.NoError(t, err, "failed to unmarshal infra manifest") {
+				return
+			}
+			assert.Equal(t, tc.expectedInfrastructure, &actualInfra)
+		})
+	}
+}
+
+type icOption func(*types.InstallConfig)
+
+type icBuildNamespace struct{}
+
+var icBuild icBuildNamespace
+
+func (icBuildNamespace) build(opts ...icOption) *types.InstallConfig {
+	ic := &types.InstallConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test-cluster",
+		},
+		BaseDomain:   "test-domain",
+		ControlPlane: &types.MachinePool{},
+	}
+	for _, opt := range opts {
+		opt(ic)
+	}
+	return ic
+}
+
+func (b icBuildNamespace) forAWS() icOption {
+	return func(ic *types.InstallConfig) {
+		if ic.Platform.AWS != nil {
+			return
+		}
+		ic.Platform.AWS = &awstypes.Platform{}
+	}
+}
+
+func (b icBuildNamespace) withServiceEndpoint(name, url string) icOption {
+	return func(ic *types.InstallConfig) {
+		b.forAWS()(ic)
+		ic.Platform.AWS.ServiceEndpoints = append(
+			ic.Platform.AWS.ServiceEndpoints,
+			awstypes.ServiceEndpoint{
+				Name: name,
+				URL:  url,
+			},
+		)
+	}
+}
+
+type infraOption func(*configv1.Infrastructure)
+
+type infraBuildNamespace struct{}
+
+var infraBuild infraBuildNamespace
+
+func (b infraBuildNamespace) build(opts ...infraOption) *configv1.Infrastructure {
+	infra := &configv1.Infrastructure{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: configv1.SchemeGroupVersion.String(),
+			Kind:       "Infrastructure",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "cluster",
+		},
+		Spec: configv1.InfrastructureSpec{
+			PlatformSpec: configv1.PlatformSpec{},
+		},
+		Status: configv1.InfrastructureStatus{
+			InfrastructureName:     "test-infra-id",
+			APIServerURL:           "https://api.test-cluster.test-domain:6443",
+			APIServerInternalURL:   "https://api-int.test-cluster.test-domain:6443",
+			ControlPlaneTopology:   configv1.HighlyAvailableTopologyMode,
+			InfrastructureTopology: configv1.HighlyAvailableTopologyMode,
```

Comment:
[govet][1]:


```
ERRO[2021-05-08T13:27:51Z] vet: pkg/asset/manifests/infrastructure_test.go:136:37: HighlyAvailableTopologyMode not declared by package v1
```

@staebler , looks like this test will need some pivots for 4.7, and I haven't been following closely enough to know what they are.

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_installer/4918/pull-ci-openshift-installer-release-4.7-govet/1391021180915290112#1:build-log.txt%3A21